tools/spectral/ipa/rulesets/IPA-114.yaml

```
@@ -32,3 +33,15 @@ rules:
     given: $.components.schemas.ApiError
     then:
       function: 'IPA114ApiErrorHasBadRequestDetail'
+  xgen-IPA-114-authenticated-endpoints-have-auth-errors:
+    description: |
+      Authenticated endpoints must define 401 and 403 responses.
+
+      ##### Implementation details
+      This rule checks that all authenticated endpoints (those without explicit 'security: []' 
+      and not containing '/unauth' in the path) include 401 and 403 responses.
+    message: '{{error}} https://mdb.link/mongodb-atlas-openapi-validation#xgen-IPA-114-authenticated-endpoints-have-auth-errors'
+    severity: warn
+    given: '$.paths[*][*]'
```

```suggestion
    given: '$.paths[*][put,patch,get,delete...]'
```
I think you need to add the methods explicitly here (similarly to a few of the IPA 117 rules), otherwise it will validate extensions as well. I can see in the results:

```
warning  xgen-IPA-114-authenticated-endpoints-have-auth-errors  Authenticated endpoint must define a 401 and 403 responses. https://mdb.link/mongodb-atlas-openapi-validation#xgen-IPA-114-authenticated-endpoints-have-auth-errors                                                                                                        paths./api/atlas/v2/alertConfigs/matchers/fieldNames.x-xgen-IPA-exception
```

---

Nice catch, addressed 👍 507 violations 🥲

---

:hidethepainharold: